app/models/search_results/comparisons.rb

```diff
@@ -5,6 +5,11 @@ module Comparisons
     def result_rows=(result_rows)
       self.result_groups = result_rows.group_by { |row| row.parent_id }
       self.result_groups.values.map! { |row| row.map! { |r| r.child_id }.compact! }
+      self.result_groups.each do |key, value|
```

Comment:
My Ruby chops aren't really good enough to know how to incorporate this into the map in a clean way. Any suggestions?

Reply:
You may include an extra check in the line above when `map` and `compact` are filtering the items.
Perhaps this can be a one liner in the form:

```ruby
self.result_groups.map! { |key, value| value.empty ? nil : value }
```

Reply:
That one liner doesn't actually modify the values of the hash, but I added one that does. I can't figure out how to modify the original map statement to do what I want, since the decision to replace the value with nil depends on the output of the map statement, not on the input. 

Let me know if what I've got here looks ok.

Reply:
You can do this:

```ruby
self.result_groups.update(self.result_group){ |key, value| value.empty ? nil : value }
```